executor/src/witgen/processor.rs

```diff
@@ -39,35 +39,35 @@ impl<'a, T: FieldElement> OuterQuery<'a, T> {
 /// This current implementation is very rudimentary and only used in the block machine
 /// to "fix" the last row. However, in the future we can generalize it to be used
 /// for general block machine or VM witness computation.
-pub struct Processor<'a, 'b, T: FieldElement, CalldataAvailable> {
+pub struct Processor<'a, 'b, 'c, T: FieldElement, CalldataAvailable> {
```

Comment:
Not sure if there is a way around this. The meaning of these lifetimes is:
- `'a`: The duration of the entire witness generation (e.g. references to identities)
- `'b`: The duration of this machine's call (e.g. the mutable references of the other machines)
- `'c`: The duration of this Processor's lifetime (e.g. the reference to the identity processor)

Reply:
Can the identity processor be passed by value (and then returned again)?

Reply:
Also shouldn't we add this (explanation of lifetimes) into the documentation?

Reply:
Yes, it used to be the case that the identity processor is owned by the processor. I can also return it from `finish()`, but I found it a bit awkward to return a third argument there. Then it also has to be returned from `BlockMachine::process`.

Also, I'm not sure it solves the lifetime issue. If `identities` and `witness_cols` are `&'b`, like before, this shortens the lifetime of `'b`, which becomes a problem when we call the processor a second time with the same identity processor.

I'll add a comment to explain the lifetimes.